drivers/mbox/mbox_nrf_bellboard_rx.c

```diff
@@ -149,6 +151,32 @@ static DEVICE_API(mbox, bellboard_rx_driver_api) = {
 		     irq_enable(DT_INST_IRQ_BY_NAME(0, name, irq));),                              \
 		    ())
 
+#else
+
+#define BELLBOARD_IRQ_FUN(name, idx)                                                               \
+	COND_CODE_1(DT_INST_IRQ_HAS_NAME(0, name), (                                               \
+		ISR_DIRECT_DECLARE(bellboard_rx_##idx##_isr)                                       \
+		{                                                                                  \
+			bellboard_rx_isr((const void *)idx);                                       \
```

Comment:
Are you sure these ISRs (the timer one too) are being inlined?  It seems sorta silly to go through the trouble of removing a memory indirection to fetch a pointer on interrupt entry just to **add** a function call (which is usually more expensive!) to the path to the handler.